src/main/java/graphql/GraphQLInstanceException.java

```diff
@@ -0,0 +1,26 @@
+package graphql;
```

Comment:
Naming is hard 

but `GraphQLInstanceException` is not grabbing me.

The `Instance` bit doesn't sit right with me'  

So I agree design sentiment but its the name that is throwing me.

Suggestions:

* GraphQLExecutionException
* GraphQLRuntimeException

Also looking at the bullet points below it seems we do have 2 classes of "runtime /execution" exceptions

One class that deal with the "schema - building+validating" and one class that deals with the "execution" of a query (after validation / parse)

So should we split them into say

* GraphqlSchemaException
* GraphqlExecutionException

Thoughts @andimarek

Reply:
I agree `Instance` might be confusing, so to recap we would have:

```
                                          GraphQLException
                                                 +
                                                 |
              +------------------------------------------------------------------------+
              |                                  |                                     |
              |                                  |                                     |
              v                                  v                                     v
  GraphQLQueryException                 GraphQLRuntimeException               AssertInternalException
                                                 +
                                                 |
                                                 |
                              +------------------------------------+
                              |                                    |
                              |                                    |
                              v                                    v
                     GraphQLExecutionException          GraphQLSchemaException
```

Legend:
* GraphQLException: top level base class to identify exception thrown by GraphQL-java in general
* GraphQLQueryException: encompasses every thing that can go wrong when attempting to execute a GraphQL request (e.g: problems with operations, variables, graphql query parsing/validation, query too complex). This exception category is used for detecting client vs server errors.
* GraphQLRuntimeException: encompasses issues emanating from the graphql application, i.e schema building/validating/wiring and at execution time such as problems coming from the data fetcher execution or type resolution execution.

I'm still very fond of the `GraphQLRuntimeException` but I can't find a better suggestion. Thoughts?

If we are good with this top level hierarchy I'll go ahead and update the PR.

Reply:
Or just collapse it out by removing the `GraphQLRuntimeException`

```
                                                 GraphQLException
                                                 +
                                                 |
              +-------------------------------------------------------------------------------------------------------+
              |                                  |                                     |                              |
              |                                  |                                     |                              |
              v                                  v                                     v                              v
  GraphQLQueryException                 GraphQLExecutionException       GraphQLExecutionException            AssertInternalException
                                             
```

Either way is fine by me.  Thoughts @andimarek ?

Reply:
I'm good with that. (`GraphQLExecutionException` shows twice in your tree, you meant `GraphQLSchemaException` but I got the message)

We could use another set of eyes on this @andimarek. What do you think?

Reply:
yes let us collapse it